### PR TITLE
feat: stricter feature flag types

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves []

### Description

We had issues with missing echo flag keys, so we are removing the option to have feature flags that are not controlled by an echo flag key.

This does not change how unleash is handled.

I have added the missing echo flag keys.

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/store/config/features.ts [hunk 1]
```diff
@@ -106,12 +106,14 @@ export const features = defineFeatures({
   },
   ARShowLinkedAccounts: {
     readyForRelease: true,
+    echoFlagKey: "ARShowLinkedAccounts",
```
Comment (pvinis) [inline src/app/store/config/features.ts:109]: adding the rest of the flags

File: src/app/store/config/features.ts [hunk 2]
```diff
@@ -3,34 +3,34 @@ import { echoLaunchJson } from "app/utils/jsonFiles"
...
+  
+  /** Whether or not to show the feature flag in the admin menu. Consider also providing a description. */
   readonly showInAdminMenu?: boolean
 }
+
 export interface FeatureDescriptorReadyForRelease {
   /**
    * Set readyForRelease to `true` when the feature is ready to be exposed outside of dev mode.
    * If an echo flag key is specified, the echo flag's value will be used after this
-   * is set to `true` and the admin ovverides won't have an impact!
```
Comment (pvinis) [inline src/app/store/config/features.ts:19]: @MounirDhahri are we sure about this? i think the admin overrides still work.

File: src/app/store/config/features.ts (hunk 1 above)
Comment (pvinis) [inline src/app/store/config/features.ts:109]:
```suggestion
```

File: src/app/store/config/features.ts [hunk 3]
```diff
@@ -152,6 +153,7 @@ export const features = defineFeatures({
   },
   AREnablePlaceholderLayoutAnimation: {
     readyForRelease: true,
+    echoFlagKey: "AREnablePlaceholderLayoutAnimation",
```
Comment (pvinis) [inline src/app/store/config/features.ts:156]:
```suggestion
```

File: src/app/store/config/features.ts [hunk 4]
```diff
@@ -112,6 +112,7 @@ export const features = defineFeatures({
   },
   ARAllowLinkSocialAccountsOnSignUp: {
     readyForRelease: true,
+    echoFlagKey: "ARAllowLinkSocialAccountsOnSignUp",
```
Comment (pvinis) [inline src/app/store/config/features.ts:115]:
```suggestion
```

File: src/app/store/config/features.ts [hunk 5]
```diff
@@ -167,12 +167,14 @@ export const features = defineFeatures({
   },
   ARShowRequestPriceEstimateBanner: {
     readyForRelease: true,
+    echoFlagKey: "ARShowRequestPriceEstimateBanner",
```
Comment (pvinis) [inline src/app/store/config/features.ts:170]:
```suggestion
```

File: src/app/store/config/features.ts [hunk 6]
```diff
@@ -173,6 +173,7 @@ export const features = defineFeatures({
   },
   ARShowMyCollectionDemandIndexHints: {
     readyForRelease: true,
+    echoFlagKey: "ARShowMyCollectionDemandIndexHints",
```
Comment (pvinis) [inline src/app/store/config/features.ts:176]:
```suggestion
```

File: src/app/store/config/features.ts (hunk 2 above)
Comment (MounirDhahri) [reply src/app/store/config/features.ts:19]: Yes, I was also surprised about it as well, when `readyForRelease` is set to `true`, the admin overrides are not taken into consideration, only echo or return `true`

**Comment by MounirDhahri** [comment]
@pvinis I created a PR yesterday that does something similar to this as a follow-up from our conversation last week when I forgot to add the `echoFlagKey` here is it https://github.com/artsy/eigen/pull/6833
The full conversation is here https://artsy.slack.com/archives/C02BAQ5K7/p1653913863784989

File: src/app/store/config/features.ts [hunk 7]
```diff
@@ -3,34 +3,34 @@ import { echoLaunchJson } from "app/utils/jsonFiles"
...
 }
 
 interface FeatureDescriptorNotReadyForRelease {
   /**
-   * Set readyForRelease to `false` when the feature is still in progress
+   * Set readyForRelease to `false` when the feature is still in progress.
    */
   readonly readyForRelease: false
+  
+  readonly echoFlagKey?: string
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/store/config/features.ts:33]: good idea 👍

**Review by MounirDhahri — COMMENTED** [review]
I only have one comment about the admin overrides thing otherwise we should be clear to merge.

File: src/app/store/config/features.ts (hunk 2 above)
Comment (pvinis) [reply src/app/store/config/features.ts:19]: but theres this https://github.com/artsy/eigen/blob/main/src/app/store/config/FeaturesModel.tsx#L39-L41 right above.. right?

**Review by MounirDhahri — APPROVED** [review]
🎉

**Pull request merged**